[@linaria/esbuild] Errors occur due to missing of loader and resolveDir

## Environment

- Linaria version: 3.0.0-beta.18
- Bundler (+ version): esbuild (0.14.22)
- Node.js version: 16.14.0
- OS: MacOS Big Sur 11.6.4

## Description

I get an error when using `@linaria/esbuild` with other esbuild plugins. The following error occurred when `@linaria/esbuild` was incorporated into Remix esbuild.

```
✘ [ERROR] Could not resolve "remix"

    route:/Users/uejima.aiji/RubymineProjects/remix-esbuild-override/examples/linaria/app/root.tsx:8:7:
      8 │ } from "remix";
        ╵        ~~~~~~~

  The plugin "linaria" didn't set a resolve directory for the file "route:/Users/uejima.aiji/RubymineProjects/remix-esbuild-override/examples/linaria/app/root.tsx", so esbuild did not search for "remix" on the file system.

 ✘ [ERROR] Could not resolve "@linaria/core"

    route:/Users/uejima.aiji/RubymineProjects/remix-esbuild-override/examples/linaria/app/routes/index.tsx:3:30:
      3 │           import { css } from "@linaria/core";
        ╵                               ~~~~~~~~~~~~~~~

  The plugin "linaria" didn't set a resolve directory for the file "route:/Users/uejima.aiji/RubymineProjects/remix-esbuild-override/examples/linaria/app/routes/index.tsx", so esbuild did not search for "@linaria/core" on the file system.

 ✘ [ERROR] Could not resolve "~/components/Paragraph"

    route:/Users/uejima.aiji/RubymineProjects/remix-esbuild-override/examples/linaria/app/routes/index.tsx:4:42:
      4 │ import { Paragraph, ParagraphStyle } from "~/components/Paragraph";
        ╵                                           ~~~~~~~~~~~~~~~~~~~~~~~~

  The plugin "linaria" didn't set a resolve directory for the file "route:/Users/uejima.aiji/RubymineProjects/remix-esbuild-override/examples/linaria/app/routes/index.tsx", so esbuild did not search for "~/components/Paragraph" on the file system.


Build failed with 3 errors:
route:/Users/uejima.aiji/RubymineProjects/remix-esbuild-override/examples/linaria/app/root.tsx:8:7: ERROR: Could not resolve "remix"
route:/Users/uejima.aiji/RubymineProjects/remix-esbuild-override/examples/linaria/app/routes/index.tsx:3:30: ERROR: Could not resolve "@linaria/core"
route:/Users/uejima.aiji/RubymineProjects/remix-esbuild-override/examples/linaria/app/routes/index.tsx:4:42: ERROR: Could not resolve "~/components/Paragraph"

```

## Solution

This problem occurs because the return value of build.onLoad does not contain loader and resolveDir. You need to add them in the two places in the link below.
https://github.com/callstack/linaria/blob/master/packages/esbuild/src/index.ts#L81
https://github.com/callstack/linaria/blob/master/packages/esbuild/src/index.ts#L101

I have already applied the RokaR patch and confirmed that this problem has been resolved.

--- a/packages/esbuild/src/index.ts
+++ b/packages/esbuild/src/index.ts
@@ -79,6 +79,8 @@ export default function linaria({
         if (!result.cssText) {
           return {
             contents: code,
+            loader,
+            resolveDir: path.basename(args.path),
           };
         }
 
@@ -99,6 +101,8 @@ export default function linaria({
           import ${JSON.stringify(cssFilename)};
           ${result.code}
           `,
+          loader,
+          resolveDir: path.basename(args.path),
         };
       });
     },

